docs/src/contributing/pull-request.md

```diff
@@ -0,0 +1,126 @@
+# The PR process for agave source code development
+
+## Preparation
+
+1. You’ll want a powerful machine for this. Building agave from sources takes a lot of RAM, make sure you have ~30GB to be sure it is enough.
+   1. Rust analyzer is quite capable of running on agave source tree, but expect high memory consumption
+2. Make a fork of anza-xyz/agave master branch under your own namespace in github. Let us assume it is called ubercoder/agave
+   1. Do not ever commit anything into the master branch of your fork. Keep it synced to upstream, this will make rebase process painless.
+
+## When not to file a PR
+
+1. Your code/comments may expose/highlight a security threat in the existing deployed instances
+
+
+## Procedure
```

Comment:
I'm thinking we could up-level this section a little bit. Doesn't feel right to get too specific with how to manage git CLI commands. Maybe we can stick to concepts - e.g. rebase from upstream and force push to private branch as opposed to `git rebase master` then `git push –force** fix\_all\_bugs`

Just a thought.. curious what others think.

Reply:
makes sense, done that.